config/config.go

```diff
@@ -1,6 +1,7 @@
 package config
 
 import (
+	"sync"
 	"time"
 
 	"github.com/sirupsen/logrus"
```

Comment:
Just curious: does "sync" also have support for spinlocks, etc. ?

Reply:
Looks like they do their own thing, which kinda looks like spinlock but with a bit of control on the 'spinning'. 
https://golang.org/src/sync/mutex.go

but then they have this bit at the end, if(race.Enabled), which ends up pointing to the 'internal' package which is a bunch of platform specific stuff. (like a BSD specific type of mutex might actually be instantiated when you're on a BSD platform). 

here's just other docs i ended up reading this morning. 
https://kernel.readthedocs.io/en/sphinx-samples/kernel-locking.html

And thanks for looking into this and merging. There are tickets here if you ever get bored enough.